test/server.test.js

```diff
@@ -1,14 +1,347 @@
-const chai = require('chai');
-const chaiHttp = require('chai-http');
-const server = require('../app/server'); // Adjust the path as needed
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+const server = require("../app/server");
 
-const should = chai.should();
+chai.should();
 chai.use(chaiHttp);
 
-// TODO: Write tests for the server
+describe("Login", () => {
+  describe("POST /api/login", function () {
+    it("Should return an access token for authentication if login credentials are valid", function (done) {
+      const login = { username: "yellowleopard753", password: "jonjon" };
 
-describe('Brands', () => {});
+      chai
+        .request(server)
+        .post("/api/login")
+        .send(login)
+        .end((err, res) => {
+          res.status.should.equal(200);
+          res.body.should.be.an("object");
+          res.body.should.have.property("token");
+          res.body.token.should.be.a("string");
+          done();
+        });
+    });
 
-describe('Login', () => {});
+    it("Should return an error message for invalid login credentials", function (done) {
+      const login = { username: "wrongUsername", password: "wrongPassword" };
 
-describe('Cart', () => {});
+      chai
+        .request(server)
+        .post("/api/login")
+        .send(login)
+        .end((err, res) => {
+          res.status.should.equal(401);
+          res.body.should.be.an("object");
+          res.body.should.have
+            .property("message")
+            .equal("Invalid login credentials");
+          done();
+        });
+    });
+  });
+});
+
+describe("Brands", () => {
+  describe("GET /api/sunglasses/brands", () => {
+    it("should return a 'Brand name required in query' error message if no brand parameter is present in query", function (done) {
+      chai
+        .request(server)
+        .get("/api/sunglasses/brands?")
+        .end((err, res) => {
+          res.status.should.equal(400);
+          res.body.should.be.an("object");
+          res.body.message.should.equal("Brand name required in query");
+          done();
+        });
+    });
+
+    it("should return an array of sunglasses for a given valid brand", function (done) {
+      const brand = "Oakley";
+
+      chai
+        .request(server)
+        .get(`/api/sunglasses/brands?brand=${brand}`)
+        .end((err, res) => {
+          res.status.should.equal(200);
+          res.body.should.be.an("array");
+          res.body.length.should.be.above(0);
```

Comment:
this is not really testing anything, if you have a bug and return a bad item for a different brand, this test will pass and you will have a bug in production. 
You should use .equals and test agains a mock.